datacube/drivers/postgis/_api.py

```diff
@@ -1474,33 +1500,38 @@ def remove_lineage_relations(self,
     def temporal_extent_by_prod(self, product_id: int) -> tuple[datetime.datetime, datetime.datetime]:
         query = self.temporal_extent_full().where(Dataset.product_ref == product_id)
         res = self._connection.execute(query)
-        return res.first()
+        tmin, tmax = res.first()
```

Comment:
I realize the existing return type on the function says tuple, but are you really guaranteed to never get `None` from `res.first()`? (And same question for next function.)

Reply:
Yeah good catch.